### ISSUE TITLE
[Feature Request] - Support m3u8 for streaming radio

### ISSUE THREAD

**Issue by adrienneteicher** [CappielloAntonio/tempo#233]
## Summary
Support m3u8 for streaming radio

## Description
Streams with m3u8 format fail with error message "unknown format"

## Use Case
Many web radio streams use this format 

## Additional context
https://mediaserviceslive.akamaized.net/hls/live/2038316/classicfmnsw/masterhq.m3u8

**Comment by CappielloAntonio** [comment]
The "Unknown format" message is not an error message. The radio stream is indeed of unknown format as we do not have the actual source. HSL streams are managed now, so it works. 

There is a problem however and it concerns the new caching management of music tracks. If enabled, the player will also try to cache the hls flow, which however takes the data on request from the server that provides it. To listen to this new stream, the streaming caching option must be disabled in the settings.

You can try it:
[app-notquitemy-debug.zip](https://github.com/CappielloAntonio/tempo/files/15447630/app-notquitemy-debug.zip)

**Comment by adrienneteicher** [comment]
Hey there.

As a layperson I was not able to follow your answer.

I can not see the streaming cahcing option in the settings.

Also, I am not sure what I should do with the apk you have provided.

Anyway. Thanks for all the work you have done. I am happy enough just using Tempo for my music.

**Comment by CappielloAntonio** [comment]
The application installed on your device at this moment does not allow you to play the radio streaming you are talking about.

I have implemented this type of playback even though a release containing it has not yet been made. The apk I have attached is a test one, in case you want to try that particular stream before the final release.

The problem I was telling you about is intrinsic to the flow type and caching algorithm. For HLS streams caching must be disabled.
